packages/airnode-node/src/api/index.ts

```diff
@@ -105,46 +101,49 @@ async function signWithTemplateId(
   );
 }
 
+type CallApiConfig = Pick<Config, 'ois' | 'apiCredentials'> &
+  Partial<Pick<Config, 'chains'>> &
+  Partial<{
+    nodeSettings: Pick<Config['nodeSettings'], 'airnodeWalletMnemonic'>;
+  }>;
+
 export interface CallApiPayload {
-  readonly config: Config;
+  readonly config: CallApiConfig;
   readonly aggregatedApiCall: AggregatedApiCall;
 }
 
 function verifySponsorWallet(payload: CallApiPayload): LogsData<ApiCallErrorResponse> | null {
   const { config, aggregatedApiCall } = payload;
+
   if (aggregatedApiCall.type !== 'regular') return null;
 
+  const { nodeSettings } = config;
+  if (!nodeSettings) {
+    const message = 'Missing config.nodeSettings object';
+    const log = logger.pend('ERROR', message);
+    return [[log], { success: false, errorMessage: message }];
+  }
```

Comment:
I believe this check won't be necessary once the types are correct as we will be sure that the field is there for the regular request as the config is validated during the deployment.